### PR TITLE
Fix Slingshot and Crosshair.

### PR DESCRIPTION
**Pull request by tclaar**
All I did was instantiate those classes, load the crosshair spritesheet, and update some references.

### CODE REVIEW

File: js/classes/entities/foreground/crosshair.js [hunk 1]
```diff
@@ -15,8 +15,7 @@ class Crosshair {
 
     update() {
         this.pos = GAME.mousePos;
-
-        if (GAME.mouseDown) {
+        if (GAME.user.aiming) {
```
Comment (calebkrauter) [inline js/classes/entities/foreground/crosshair.js:18]: Thanks for doing the refactor here!

File: js/classes/game_engine.js [hunk 2]
```diff
@@ -123,14 +123,14 @@ class GameEngine {
...
+        CANVAS.addEventListener("mousedown", (e) => {
+            this.user.firing = false;
+            this.user.aiming = true;
         });
 
-        CANVAS.addEventListener("mouseUp", (e) => {
-            this.user.aiming = true;
-            this.user.firing = false;
+        CANVAS.addEventListener("mouseup", (e) => {
+            this.user.aiming = false;
```
Comment (calebkrauter) [inline js/classes/game_engine.js:132]: It's weird to me that everything was reversed prior to your changes.

File: js/classes/zone.js [hunk 3]
```diff
@@ -26,6 +26,11 @@ class Zone {
         this.load = () => {
             // Clear all entities from previous Zone.
             GAME.clearEntities();
+
+            // Add entities essential to any Zone.
```
Comment (calebkrauter) [inline js/classes/zone.js:30]: That's a good call.

**Review by calebkrauter — APPROVED** [review]
Thanks @tclaar for fixing the slingshot! Not sure if it broke when I did refactoring or prior but it is odd to me that everything was backwards. Regardless the slingshot was not added as an entity when I was working on it. Nice work fixing it up, glad it wasn't too intensive!

File: js/classes/game_engine.js (hunk 2 above)
Comment (tclaar) [reply js/classes/game_engine.js:132]: The way you had it made intuitive sense - you were setting firing to true when the mouse was down and setting aiming to true when the mouse was up. It's just that we want our slingshot to work in the opposite way.

File: js/classes/zone.js (hunk 3 above)
Comment (tclaar) [reply js/classes/zone.js:30]: It's possible there is a better place to do this, @B1GD3VD0G would know better than me. It made sense to me to do it here though.

File: js/classes/entities/foreground/crosshair.js (hunk 1 above)
Comment (NathanHinthorne) [reply js/classes/entities/foreground/crosshair.js:18]: Thanks for using the new naming convention

**Review by NathanHinthorne — APPROVED** [review]

**Pull request merged**